src/Core/Services/EnvironmentService.cs

```diff
@@ -113,20 +169,19 @@ public async Task<EnvironmentUrlData> SetUrlsAsync(EnvironmentUrlData urls)
             IconsUrl = urls.Icons;
             NotificationsUrl = urls.Notifications;
             EventsUrl = urls.Events;
+            _apiService.SetUrls(urls);
+            return urls;
+        }
 
-            var envUrls = new EnvironmentUrls();
-            if (!string.IsNullOrWhiteSpace(BaseUrl))
-            {
-                envUrls.Base = BaseUrl;
-            }
-            else
-            {
-                envUrls.Api = ApiUrl;
-                envUrls.Identity = IdentityUrl;
-                envUrls.Events = EventsUrl;
-            }
-
-            _apiService.SetUrls(envUrls);
+        private EnvironmentUrlData FormatUrls(EnvironmentUrlData urls)
+        {
+            urls.Base = FormatUrl(urls.Base);
+            urls.WebVault = FormatUrl(urls.WebVault);
+            urls.Api = FormatUrl(urls.Api);
+            urls.Identity = FormatUrl(urls.Identity);
+            urls.Icons = FormatUrl(urls.Icons);
+            urls.Notifications = FormatUrl(urls.Notifications);
+            urls.Events = FormatUrl(urls.Events);
             return urls;
         }
```

Comment:
🎨 I think both `FormatUrls(...)` methods should be moved inside `EnvironmentUrlData` given that IMO the service shouldn't be responsible of formatting it but the object itself; additionally it improves maintainability if another url is added to the `EnvironementUrlData`, so the method is just in there to be seen by the developer and update it accordingly.